### PR TITLE
compiler: Patch generate_implicit for corner cases

### PR DESCRIPTION
**Pull request by FabioLuporini**
E.g. dummy operators with a parallel time dimensions

### CODE REVIEW

File: devito/passes/clusters/implicit.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 """
-Passes to gather and form implicit equations from MultiSubDomains.
+Passes to gather and form implicit equations from DSL abstractions.
```
Comment (georgebisbas) [inline devito/passes/clusters/implicit.py:2]: + (e.g. MultiSubDomains...)

File: tests/test_subdomains.py [hunk 2]
```diff
@@ -622,3 +622,32 @@ class Dummy(SubDomainSet):
 
         assert_structure(op, ['t,n', 't,n,xi_n0_blk0,yi_n0_blk0,xi_n,yi_n,zi_n'],
                          't,n,xi_n0_blk0,yi_n0_blk0,xi_n,yi_n,zi_n')
+
+    def test_sequential_implicit(self):
+        """
+        Make sure implicit dimensions define sequential iteration spaces. This is for
```
Comment (georgebisbas) [inline tests/test_subdomains.py:628]: Maybe refactor comment as discussed offline.

File: tests/test_subdomains.py [hunk 3]
```diff
@@ -622,3 +622,32 @@ class Dummy(SubDomainSet):
...
+        performance and potentially for correctness too (e.g., canonical openmp loops
+        forbid subiterators, which could potentially be required by a MultiSubDomain).
+        """
+
+        class Dummy(SubDomainSet):
+            name = 'dummy'
+
+        dummy = Dummy(N=0, bounds=[(), (), (), (), (), ()])
```
Comment (georgebisbas) [inline tests/test_subdomains.py:636]: Line 636 to 645 too many blank lines?

File: tests/test_subdomains.py [hunk 4]
```diff
@@ -622,3 +622,32 @@ class Dummy(SubDomainSet):
...
+        grid = Grid(shape=(10, 10, 10), subdomains=(dummy,))
+
+        f = TimeFunction(name='f', grid=grid, save=10)
+
+        eqn = Eq(f, 1., subdomain=grid.subdomains['dummy'])
+
+        op = Operator(eqn)
+
+        iterations = FindNodes(Iteration).visit(op)
+        assert len(iterations) == 5
```
Comment (georgebisbas) [inline tests/test_subdomains.py:647]: assert length is redundant since the assignment takes place in the next line?
You can also use assert_structure

**Review by georgebisbas — COMMENTED** [review]
Minor comments, otherwise GTG.

File: tests/test_subdomains.py (hunk 3 above)
Comment (FabioLuporini) [reply tests/test_subdomains.py:636]: It follows the grouping logic we always use

Comment (FabioLuporini) [reply tests/test_subdomains.py:636]: the one for tests I mean

**Review by georgebisbas — APPROVED** [review]

**Pull request merged**